sdk/interchain/signature-aggregator.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved
+// See the file LICENSE for licensing terms.
+package interchain
+
+import (
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/message"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	apiConfig "github.com/ava-labs/awm-relayer/config"
+	"github.com/ava-labs/awm-relayer/peers"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/aggregator"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/config"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/metrics"
+	awmTypes "github.com/ava-labs/awm-relayer/types"
+	awmUtils "github.com/ava-labs/awm-relayer/utils"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	DefaultQuorumPercentage   = uint64(67)
+	DefaultSignatureCacheSize = uint64(1024 * 1024)
+)
+
+var etnaTime = time.Unix(0, 0)
+
+type SignatureAggregator struct {
+	subnetID         ids.ID
+	quorumPercentage uint64
+	aggregator       *aggregator.SignatureAggregator
+}
+
+func CreateAppRequestNetwork(network models.Network, logLevel logging.Level) (peers.AppRequestNetwork, error) {
```

Comment:
nice to see a contribution to SDK in a CLI PR +1

Reply:
gonna resolve this one. thank you :)